components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterEndpoint.java

```diff
@@ -140,30 +149,87 @@ protected void doInit() throws Exception {
+                aggregateExecutorService = camelContext.getExecutorServiceManager()
+                        .newScheduledThreadPool(this, "DynamicRouter-AggregateTask", 0);
+            }
+            AggregationStrategy aggregationStrategy = determineAggregationStrategy(camelContext);
+            DynamicRouterMulticastProcessor processor = processorFactorySupplier.get()
+                    .getInstance("DynamicRouterMulticastProcessor-" + configuration.getChannel(), camelContext, null,
+                            configuration.getRecipientMode(),
+                            configuration.isWarnDroppedMessage(), filterProcessorFactorySupplier, producerCache,
+                            aggregationStrategy, configuration.isParallelProcessing(),
+                            determineExecutorService(camelContext), configuration.isShutdownExecutorService(),
+                            configuration.isStreaming(), configuration.isStopOnException(), timeout,
+                            determineOnPrepare(camelContext), configuration.isShareUnitOfWork(),
+                            configuration.isParallelAggregate());
+            processor.setErrorHandler(errorHandler);
+            processor.setAggregateExecutorService(aggregateExecutorService);
+            processor.setIgnoreInvalidEndpoints(configuration.isIgnoreInvalidEndpoints());
+            processor.setId(getId());
+            processor.setRouteId(routeId);
+            ServiceHelper.startService(aggregationStrategy, producerCache, processor);
             component.addRoutingProcessor(configuration.getChannel(), processor);
         }
     }
 
+    protected ExecutorService determineExecutorService(CamelContext camelContext) {
+        ExecutorService executorService = null;
+        if (ObjectHelper.isNotEmpty(configuration.getExecutorService())) {
+            executorService = camelContext.getExecutorServiceManager()
+                    .newThreadPool(this, "@RecipientList", configuration.getExecutorService());
+        }
+        if (configuration.isParallelProcessing() && configuration.getExecutorService() == null) {
+            // we are running in parallel, so we need a thread pool
+            executorService = camelContext.getExecutorServiceManager()
+                    .newDefaultThreadPool(this, "@RecipientList");
+        }
+        return executorService;
+    }
+
+    protected AggregationStrategy determineAggregationStrategy(CamelContext camelContext) {
+        AggregationStrategy aggregationStrategy = new UseLatestAggregationStrategy();
+        if (ObjectHelper.isNotEmpty(configuration.getAggregationStrategy())) {
+            aggregationStrategy = CamelContextHelper.mandatoryLookup(camelContext,
+                    configuration.getAggregationStrategy(), AggregationStrategy.class);
+        }
+        return aggregationStrategy;
```

Comment:
This is creating an `UseLatestAggregationStrategy` object unnecessarily, which can add overhead to the GC. Try something like this:

```
        if (ObjectHelper.isNotEmpty(configuration.getAggregationStrategy())) {
            return CamelContextHelper.mandatoryLookup(camelContext,
                    configuration.getAggregationStrategy(), AggregationStrategy.class);
        }
        // only create a new one if necessary
        return new UseLatestAggregationStrategy()
```

Reply:
Thanks.  I will make the change as requested.